src/protocol/primitives/int16.ts

```diff
@@ -0,0 +1,25 @@
+import type { ReadBuffer, Serializable, WriteBuffer } from '../serialization';
+import { checkValueIsInRange } from './utils';
+
+/**
+ * Represents an integer between -2^15 and 2^15-1 inclusive.
+ * The values are encoded using two bytes in network byte order (big-endian).
+ *
+ * @see https://kafka.apache.org/protocol.html#protocol_types
+ */
+export class Int16 implements Serializable {
+  public static readonly MAX_VALUE = 32767;
```

Comment:
Is there any reason to keep this public?

Reply:
Leave this public and check if it is used in the future. If not change the visibility to private.

Reply:
Resolved in https://github.com/trisquareeu/kafkit/pull/1#discussion_r1260284907